File: frontend/types/applicant.ts
export interface ApplicantInput {
  revolving_utilization: number;
  age: number;
  times_30_59_days_late: number;
  debt_ratio: number;
  monthly_income: number;
  open_credit_lines: number;
  times_90_days_late: number;
  real_estate_loans: number;
  times_60_89_days_late: number;
  dependents: number;
}

export const FIELD_METADATA: Record<
  keyof ApplicantInput,
  { label: string; tooltip: string; unit?: string }
> = {
  revolving_utilization: {
    label: "Credit Utilization",
    tooltip: "Total balance on credit cards divided by total credit limits. Values above 1.0 indicate over-limit usage.",
    unit: "ratio",
  },
  age: {
    label: "Applicant Age",
    tooltip: "Age of the borrower in years.",
    unit: "years",
  },
  times_30_59_days_late: {
    label: "30–59 Day Late Payments",
    tooltip: "Number of times the borrower has been 30–59 days past due in the last 2 years.",
  },
  debt_ratio: {
    label: "Debt Ratio",
    tooltip: "Monthly debt payments divided by monthly gross income. A ratio above 0.43 is generally considered high.",
    unit: "ratio",
  },
  monthly_income: {
    label: "Monthly Income",
    tooltip: "Gross monthly income in USD.",
    unit: "USD",
  },
  open_credit_lines: {
    label: "Open Credit Lines",
    tooltip: "Total number of open loans and lines of credit.",
  },
  times_90_days_late: {
    label: "90+ Day Late Payments",
    tooltip: "Number of times the borrower has been 90 or more days past due. Strong predictor of default.",
  },
  real_estate_loans: {
    label: "Real Estate Loans",
    tooltip: "Number of mortgage and real estate loans including home equity lines.",
  },
  times_60_89_days_late: {
    label: "60–89 Day Late Payments",
    tooltip: "Number of times the borrower has been 60–89 days past due in the last 2 years.",
  },
  dependents: {
    label: "Number of Dependents",
    tooltip: "Number of dependents in the family, excluding the borrower.",
  },
};

File: backend/app/schemas/applicant.py
from pydantic import BaseModel, Field


class ApplicantInput(BaseModel):
    revolving_utilization: float = Field(
        ..., ge=0.0, le=20.0, description="Total balance on credit cards / sum of credit limits"
    )
    age: int = Field(..., ge=18, le=110, description="Age of borrower in years")
    times_30_59_days_late: int = Field(
        ..., ge=0, le=98, description="Times borrower has been 30-59 days past due (not worse)"
    )
    debt_ratio: float = Field(
        ..., ge=0.0, le=500.0, description="Monthly debt payments / monthly gross income"
    )
    monthly_income: float = Field(..., ge=0.0, le=10_000_000.0, description="Monthly income in USD")
    open_credit_lines: int = Field(..., ge=0, le=100, description="Number of open loans and lines of credit")
    times_90_days_late: int = Field(..., ge=0, le=98, description="Times borrower has been 90+ days late")
    real_estate_loans: int = Field(
        ..., ge=0, le=50, description="Number of mortgage and real estate loans"
    )
    times_60_89_days_late: int = Field(
        ..., ge=0, le=98, description="Times borrower has been 60-89 days past due (not worse)"
    )
    dependents: int = Field(..., ge=0, le=20, description="Number of dependents excluding borrower")

    model_config = {
        "json_schema_extra": {
            "example": {
                "revolving_utilization": 0.45,
                "age": 42,
                "times_30_59_days_late": 1,
                "debt_ratio": 0.32,
                "monthly_income": 5800.0,
                "open_credit_lines": 8,
                "times_90_days_late": 0,
                "real_estate_loans": 1,
                "times_60_89_days_late": 0,
                "dependents": 2,
            }
        }
    }
